test/buy_and_activate.js

```diff
@@ -76,6 +76,17 @@ contract('Court presale and activate wrapper', ([_, owner, provider, juror1]) =>
         cpa = await CourtPresaleActivate.new(owner, bondedToken.address, registry.address, presale.address, uniswapFactory.address, { from: owner })
       })
 
+      const checkFinalBalances = async (amount, activate, initialActiveAmount) => {
+        const bondedTokensToGet = await presale.contributionToTokens(amount)
+        const finalActiveAmount = (await registry.balanceOf(juror1))[0]
+        const finalUserBalance = await bondedToken.balanceOf(juror1)
+        const finalExpectedAmount = initialActiveAmount.add(bondedTokensToGet)
+        const expectedActiveAmount = activate ? finalExpectedAmount : bn(0)
+        const expectedUserBalance = finalExpectedAmount.sub(expectedActiveAmount)
```

Comment:
I was expecting to find another `activate ? <a> : <b>` case here, since this should also change based on whether activation happens.

I realize it does the same here, but wondering if it'd be clearer to explicitly state the conditional.

Reply:
Really? I did this on purpose, because it looks way better to me: active balance and user balance are complementary, so tokens are either activated or either in user's account. The whole point of this PR was to address offline comment of "staking but not activating makes 0 sense", and this way it reflects exactly that.

Reply:
The `finalExpectedAmount.sub(expectedActiveAmount)` raises some small alarms in me because we're doing computation on a value that should be known ahead of time.

I wonder if this would be most clear:

```js
if (activate) {
  expectedActiveAmount = finalExpectedAmount
  expectedUserBalance = bn(0)
} else {
  expectedActiveAmount = bn(0)
  expectedUserBalance = finalExpectedAmount
}
```